### PR TITLE
bugfix: Disallow granting health permissions to app not handling the required intents in UI as well

### PR DESCRIPTION
**Pull request by quh4gko8**

### CODE REVIEW

File: PermissionController/src/com/android/permissioncontroller/permission/utils/HealthPermissionUtils.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import java.util.List;
+
+public class HealthPermissionUtils {
+
+    public static boolean isHealthPermissionAndShowing(@NonNull String permGroupName) {
+        return Utils.isHealthPermissionUiEnabled() && HEALTH_PERMISSION_GROUP.equals(permGroupName);
```
Comment (muhomorr) [inline PermissionController/src/com/android/permissioncontroller/permission/utils/HealthPermissionUtils.java:20]: `isHealthPermissionUiEnabled()` is much slower than `String.equals()` check. It's better to call `String.equals()` first to make this method faster in the common case.

File: PermissionController/src/com/android/permissioncontroller/permission/utils/HealthPermissionUtils.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        List<ResolveInfo> ris = context.getPackageManager().queryIntentActivitiesAsUser(
+                new Intent(Intent.ACTION_VIEW_PERMISSION_USAGE)
+                        .addCategory(HealthConnectManager.CATEGORY_HEALTH_PERMISSIONS)
+                        .setPackage(pkgName),
+                PackageManager.ResolveInfoFlags.of(PackageManager.MATCH_ALL),
+                user
+        );
+
+        if (ris.isEmpty()) {
+            Log.e(HealthPermissionUtils.class.getSimpleName(), "Health connect permissions removed for " + pkgName + ". "
```
Comment (muhomorr) [inline PermissionController/src/com/android/permissioncontroller/permission/utils/HealthPermissionUtils.java]: Outdated comment

File: PermissionController/src/com/android/permissioncontroller/permission/utils/HealthPermissionUtils.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import android.os.UserHandle;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import java.util.List;
+
+public class HealthPermissionUtils {
+
+    public static boolean isHealthPermissionAndShowing(@NonNull String permGroupName) {
```
Comment (muhomorr) [inline PermissionController/src/com/android/permissioncontroller/permission/utils/HealthPermissionUtils.java:19]: Method name is confusing

**Review by muhomorr — CHANGES_REQUESTED** [review]

File: PermissionController/src/com/android/permissioncontroller/permission/utils/HealthPermissionUtils.java (hunk 1 above)
Comment (quh4gko8) [reply PermissionController/src/com/android/permissioncontroller/permission/utils/HealthPermissionUtils.java:20]: Resolved in current revision

File: PermissionController/src/com/android/permissioncontroller/permission/utils/HealthPermissionUtils.java (hunk 2 above)
Comment (quh4gko8) [reply PermissionController/src/com/android/permissioncontroller/permission/utils/HealthPermissionUtils.java]: Updated with more appropriate comment or log message

File: PermissionController/src/com/android/permissioncontroller/permission/utils/HealthPermissionUtils.java (hunk 3 above)
Comment (quh4gko8) [reply PermissionController/src/com/android/permissioncontroller/permission/utils/HealthPermissionUtils.java:19]: Changed in the latest revision, unsure if that's clear enough

**Comment by muhomorr** [comment]
Commit title and PR title are incorrect, health permissions are already not grantable to such apps. This is just a UI bugfix.

**Comment by quh4gko8** [comment]
Closing in favor of https://github.com/GrapheneOS/platform_packages_modules_Permission/pull/38

**Pull request closed without merging**